usecases/auth/authorization/rbac/authorizer.go

```diff
@@ -51,19 +58,25 @@ func (m *manager) authorize(principal *models.Principal, verb string, skipAudit
 			return err
 		}
 
-		logger.WithFields(logrus.Fields{
-			"resources": prettyPermissionsResources(perm),
-			"results":   prettyStatus(allowed),
+		// Store the results for this iteration
+		permResults = append(permResults, logrus.Fields{
+			"resource": prettyPermissionsResources(perm),
+			"results":  prettyStatus(allowed),
 		})
 
-		if !skipAudit {
-			logger.Info()
-		}
-
 		if !allowed {
+			if !skipAudit {
+				logger.WithField("permissions", permResults).Error("authorization denied")
+			}
 			return fmt.Errorf("rbac: %w", errors.NewForbidden(principal, prettyPermissionsActions(perm), prettyPermissionsResources(perm)))
 		}
 	}
+
+	// Log all results at once if audit is enabled
+	if !skipAudit {
+		logger.WithField("permissions", permResults).Info()
+	}
```

Comment:
thought: Do you think it would make sense to make the logger "audit-aware"?
e.g. `logger = logger.WithContext(/* wrapped ctx with skipAudit=true */)` plus [a simple custom formatter](https://github.com/sirupsen/logrus?tab=readme-ov-file#formatters) that will not log stuff `if SkipAudit(entry.Context)`.

I'm asking because it seems that `skipAudit` is only being passed around to silence the logger, i.e. it doesn't have any effect on the authorization logic.

Also, idk how many places would benefit from this toggle, like maybe it's only used here and in 1 other function.

Reply:
nice idea, but actually we may migrate from logrus soon. in general we don't support audit logging yt, but this was important for RBAC it won't be needed in other places yet, but we can discover it down the road